### PR TITLE
Monthly Report API

### PR DESCRIPTION
**Pull request by Asna-A**

### CODE REVIEW

File: src/services/WolfDen.API/Controllers/Attendence/Attendence.cs [hunk 1]
```diff
@@ -35,5 +37,15 @@ public async Task<IResult> GeneratePdf([FromQuery]DailyDetailsPdf DailyDetailspd
             var pdf = document.GeneratePdf();
             return Results.File(pdf, "application/pdf", "DailyReport.pdf");
         }
+
+        [HttpGet("monthly-report")]
+        public async Task<IActionResult> GenerateMonthlyReport([FromQuery]MonthlyReportQuery MonthlyReportQuery, CancellationToken cancellationToken)
+        {
+            var monthlyReport= await _mediator.Send(MonthlyReportQuery, cancellationToken);
```
Comment (GSARYAPUTHRAN) [inline src/services/WolfDen.API/Controllers/Attendence/Attendence.cs:44]: try to use type insetad of var

File: src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendance/DailyDetailsQueryHandler.cs [hunk 2]
```diff
@@ -38,38 +39,47 @@ public async Task<DailyAttendanceDTO> Handle(DailyDetails request, CancellationT
...
-                            {
-                                AttendanceStatus attendanceStatusId = AttendanceStatus.WFH;
-                                notPresentDay.AttendanceStatusId = attendanceStatusId;
-                            }
-                            else
-                            {
-                                AttendanceStatus attendanceStatusId = AttendanceStatus.RestrictedHoliday;
-                                notPresentDay.AttendanceStatusId = attendanceStatusId;
-                            }
+                            AttendanceStatus attendanceStatusId = AttendanceStatus.RestrictedHoliday;
```
Comment (GSARYAPUTHRAN) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendance/DailyDetailsQueryHandler.cs:50]: cant we just assign notPresentDay.AttendanceStatusId = AttendanceStatus.RestrictedHoliday;

File: src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendance/DailyDetailsQueryHandler.cs [hunk 3]
```diff
@@ -38,38 +39,47 @@ public async Task<DailyAttendanceDTO> Handle(DailyDetails request, CancellationT
...
                     }
                 }
                 else
                 {
-                    AttendanceStatus attendanceStatusId = AttendanceStatus.Absent;
-                    notPresentDay.AttendanceStatusId = attendanceStatusId;
+                    LeaveRequest leave = await _context.LeaveRequests.Where(x => x.EmployeeId == request.EmployeeId && x.FromDate <= request.Date && request.Date <= x.ToDate && x.LeaveRequestStatusId == LeaveRequestStatus.Approved).Include(x => x.LeaveType).FirstOrDefaultAsync(cancellationToken);
+                    if (leave is null)
+                    {
+                        AttendanceStatus attendanceStatusId = AttendanceStatus.Absent;
```
Comment (GSARYAPUTHRAN) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendance/DailyDetailsQueryHandler.cs:60]: no need of this variable

File: src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendance/DailyDetailsQueryHandler.cs [hunk 4]
```diff
@@ -38,38 +39,47 @@ public async Task<DailyAttendanceDTO> Handle(DailyDetails request, CancellationT
...
+                        notPresentDay.AttendanceStatusId = attendanceStatusId;
+                    }
+                    else
+                    {
+                        LeaveType leaveType = await _context.LeaveType.FirstOrDefaultAsync(x => x.Id == leave.TypeId);
+                        if (leaveType.LeaveCategoryId == LeaveCategory.WorkFromHome)
+                        {
+                            AttendanceStatus attendanceStatusId = AttendanceStatus.WFH;
```
Comment (GSARYAPUTHRAN) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendance/DailyDetailsQueryHandler.cs:68]: no need of this variable

File: src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendance/DailyDetailsQueryHandler.cs [hunk 5]
```diff
@@ -38,38 +39,47 @@ public async Task<DailyAttendanceDTO> Handle(DailyDetails request, CancellationT
...
+                            notPresentDay.AttendanceStatusId = attendanceStatusId;
+                        }
+                        else
+                        {
+                            AttendanceStatus attendanceStatusId = AttendanceStatus.Absent;
```
Comment (GSARYAPUTHRAN) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendance/DailyDetailsQueryHandler.cs:73]: no need of this variable

File: src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendanceReport/DailyDetailsPdfQueryHandler.cs [hunk 6]
```diff
@@ -38,38 +39,47 @@ public async Task<DailyAttendanceDTO> Handle(DailyDetailsPdf request, Cancellati
...
-                                AttendanceStatus attendanceStatusId = AttendanceStatus.WFH;
-                                notPresentDay.AttendanceStatusId = attendanceStatusId;
-                            }
-                            else
-                            {
-                                AttendanceStatus attendanceStatusId = AttendanceStatus.RestrictedHoliday;
-                                notPresentDay.AttendanceStatusId = attendanceStatusId;
-                            }
-                        }
+                            AttendanceStatus attendanceStatusId = AttendanceStatus.RestrictedHoliday;
```
Comment (GSARYAPUTHRAN) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendanceReport/DailyDetailsPdfQueryHandler.cs:50]: no need of this variable

File: src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendanceReport/DailyDetailsPdfQueryHandler.cs [hunk 7]
```diff
@@ -38,38 +39,47 @@ public async Task<DailyAttendanceDTO> Handle(DailyDetailsPdf request, Cancellati
...
                     }
                 }
                 else
                 {
-                    AttendanceStatus attendanceStatusId = AttendanceStatus.Absent;
-                    notPresentDay.AttendanceStatusId = attendanceStatusId;
+                    LeaveRequest leave = await _context.LeaveRequests.Where(x => x.EmployeeId == request.EmployeeId && x.FromDate <= request.Date && request.Date <= x.ToDate && x.LeaveRequestStatusId == LeaveRequestStatus.Approved).Include(x => x.LeaveType).FirstOrDefaultAsync(cancellationToken);
+                    if (leave is null)
+                    {
+                        AttendanceStatus attendanceStatusId = AttendanceStatus.Absent;
```
Comment (GSARYAPUTHRAN) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendanceReport/DailyDetailsPdfQueryHandler.cs:60]: no need of this variable

File: src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendanceReport/DailyDetailsPdfQueryHandler.cs [hunk 8]
```diff
@@ -38,38 +39,47 @@ public async Task<DailyAttendanceDTO> Handle(DailyDetailsPdf request, Cancellati
...
+                        notPresentDay.AttendanceStatusId = attendanceStatusId;
+                    }
+                    else
+                    {
+                        LeaveType leaveType = await _context.LeaveType.FirstOrDefaultAsync(x => x.Id == leave.TypeId);
+                        if (leaveType.LeaveCategoryId == LeaveCategory.WorkFromHome)
+                        {
+                            AttendanceStatus attendanceStatusId = AttendanceStatus.WFH;
```
Comment (GSARYAPUTHRAN) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendanceReport/DailyDetailsPdfQueryHandler.cs:68]: no need of this variable

File: src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendanceReport/DailyDetailsPdfQueryHandler.cs [hunk 9]
```diff
@@ -38,38 +39,47 @@ public async Task<DailyAttendanceDTO> Handle(DailyDetailsPdf request, Cancellati
...
+                            notPresentDay.AttendanceStatusId = attendanceStatusId;
+                        }      
+                        else
+                        {
+                            AttendanceStatus attendanceStatusId = AttendanceStatus.Absent;
```
Comment (GSARYAPUTHRAN) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendanceReport/DailyDetailsPdfQueryHandler.cs:73]: no need of this variable

**Review by GSARYAPUTHRAN — APPROVED** [review]

File: src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendance/DailyDetailsQueryHandler.cs [hunk 10]
```diff
@@ -33,53 +34,53 @@ public async Task<DailyAttendanceDTO> Handle(DailyDetails request, CancellationT
...
-                    AttendanceStatus attendanceStatusId = AttendanceStatus.Absent;
-                    notPresentDay.AttendanceStatusId = attendanceStatusId;
+                    LeaveRequest leave = await _context.LeaveRequests.Where(x => x.EmployeeId == request.EmployeeId && x.FromDate <= request.Date && request.Date <= x.ToDate && x.LeaveRequestStatusId == LeaveRequestStatus.Approved).Include(x => x.LeaveType).FirstOrDefaultAsync(cancellationToken);
+                    if (leave is null)
+                    {
+                        notPresentDay.AttendanceStatusId = AttendanceStatus.Absent;
+                    }
+                    else
+                    {
+                        LeaveType leaveType = await _context.LeaveType.FirstOrDefaultAsync(x => x.Id == leave.TypeId);
```
Comment (anilkumar-geeky) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendance/DailyDetailsQueryHandler.cs:61]: use FindAsync if the data is fetched using primary key

File: src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendance/DailyDetailsQueryHandler.cs [hunk 11]
```diff
@@ -90,7 +91,6 @@ public async Task<DailyAttendanceDTO> Handle(DailyDetails request, CancellationT
              }).ToListAsync(cancellationToken);
              if(attendence is not null)
             attendence.DailyLog = attendenceRecords;
-            await _context.SaveChangesAsync(cancellationToken);
```
Comment (anilkumar-geeky) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/DailyAttendance/DailyDetailsQueryHandler.cs:93]: why?

**Review by GSARYAPUTHRAN — APPROVED** [review]

**Pull request merged**